### DISCUSSION TITLE
Normalize offset/duration/for text and editor behaviors

### DISCUSSION THREAD

**Discussion by niemyjski** [home-assistant/frontend#7814] [Automations / scripts / scenes]
Currently the automation editor uses different text and editors for time based offsets / criteria as shown below. It would be good to

1. Normalize the `duration` and `for` text to be consistent.
2. Use the same ui editor to manage the `offset`, `duration` and `for` fields.

![image](https://user-images.githubusercontent.com/1020579/100220355-da97db00-2edc-11eb-8af6-0a98150e969b.png)

Additionally this would make it easier to think about offsets especially all users. We could even take it a step further and add drop down to the editor for -+ offset with useful text so you don't have to think about date time math.

**Comment by KTibow** [comment]
> so you don't have to think about date time math.

Could you clarify this?

**Reply by niemyjski** [reply]
in programming the only thing harder than working with dates/time zones is cache invalidation. How would I know if a field supports an offset without looking at the documentation and then I have to really think about it.


In the middle image below is my `+` offset before or after or is it `-`. I know the answer but many people don't and have to really think about it. I sometimes still have to think about it too. Using verbose language like 15 minutes (`00:15:00`) before (`-`) sunrise or 15 minutes (`00:15:00`) before (`+`) sunset would go a long ways. So if these pickers were normalized to not be a text input box, and then a dropdown for `before`, `<empty>` and `after` would do wonders!
![image](https://user-images.githubusercontent.com/1020579/100543868-24076380-3218-11eb-8457-10565b610aa0.png)
